File: 1/1.py
def get_lists() -> tuple[list[int], list[int]]:
    with open("./input.txt", "r") as file:
        lines = file.readlines()

    l1, l2 = zip(*[tuple(int(i) for i in line.split()) for line in lines])
    return sorted(l1), sorted(l2)


def part_1() -> int:
    l1, l2 = get_lists()
    zipped = zip(l1, l2)
    return sum(abs(a - b) for a, b in zipped)


def part_2() -> int:
    l1, l2 = get_lists()
    return sum(Litm * l2.count(Litm) for Litm in l1)


def main():
    print(part_1())
    print(part_2())


if __name__ == "__main__":
    main()

File: src/2024/1/1.kt
import java.io.File
import kotlin.math.abs

fun getLists(): Pair<List<Int>, List<Int>> {
    val lines = File("/Users/olivia/projects/advent/src/1/input.txt").readLines()

    val (l1, l2) = lines.map { line ->
        val (left, right) = line.split(Regex("\\s+")).map { it.toInt() }
        left to right
    }.unzip()

    return l1.sorted() to l2.sorted()
}

fun part1(l1: List<Int>, l2: List<Int>): Int {
    return l1.zip(l2).sumOf { (a, b) -> abs(a - b) }
}

fun part2(l1: List<Int>, l2: List<Int>): Int {
    return l1.sumOf { lItem -> lItem * l2.count { it == lItem } }
}

fun day1() {
    val (l1, l2) = getLists()
    part1(l1, l2).also(::println)
    part2(l1, l2).also(::println)
}
